### DISCUSSION TITLE
Quat Slerp - No examples?

### DISCUSSION THREAD

**Discussion by laultman** [StereoKit/StereoKit#623] [Q&A] [answered]
Trying to figure out how to use Quat Slerp and other functions in SK. I could not find any demonstration in the SK demo for this function. I am trying to transform and rotate model based on where the user is in an environment, sort of like "follow-me" concept.

**Comment by maluoi** [comment]
[Slerp](https://stereokit.net/Pages/StereoKit/Quat/Slerp.html) is just a higher quality lerp for Quaternions. It blends between two different Quaternions while preserving some of the spherical properties! You'd use it kinda like this:

```csharp
Quat  start       = ...;
Quat  destination = ...;
float percentage  = ...;
Quat  current = Quat.Slerp(start, destination, percentage);
```

Or for like a smoothed chasing sort of behavior:
```csharp
Quat  destination = ...;
Quat  current     = ...;
float followPower = 16; // Somewhat arbitrary, higher is snappier

current = Quat.Slerp(current, destination, followPower * Time.Stepf);
```

**Comment by laultman** [comment]
In the first suggestion above are start and destination set "outside" the Update loop and just the percentage changing? In the second suggestion is seems that destination is maybe some point in space relative to the current head position? I guess I'm trying to figure out how it works in the bigger picture of the Update loop. In my experiments it doesn't do anything or I can't find where it went.
For my experiment to learn the Slerp concept, I have a sphere (earth) and I want to rotate it smoothly about its axis (Y) and I want it to rotate about the Sun. I have a method that finds a point on the great circle of the earth orbit about the sun. Each point is some distance in degrees (12) that need to be smoothed from each point. So I started with the idea of using slerp to rotate the planet. We know the earth rotates through 360 degrees. It starts at 0 and resets to 0 at 360 (destination) degrees. 
I have the following code in the Update method where start and destination are fixed values and the percentage is changing. Nothing happens.
`			Quat qR = Quat.Slerp(Quat.FromAngles(0, 0, 23.0f), Quat.FromAngles(0, 360.0f, 23.0f), pR / 360.0f);
			pR = (pR > 360) ? 0 : pR += .5f;
			earthModel.Draw(Matrix.TRS(new Vec3(0, -.2f, -1.2f), qR, earthScale));
`
I clearly do not understand the principal of the slerp. It appears to me that it should be somewhere between these rotations at the amount of the percentage.

**Reply by maluoi** [reply, marked as answer]
Here's a complete application for context:
```csharp
using System;
using StereoKit;

SK.Initialize();

Quat   destination = Quat.Identity;
Quat   current     = destination;
Random rand        = new();

SK.Run(() => {
	if (Input.Key(Key.Space).IsJustActive()) {
		destination = Quat.FromAngles(
			rand.NextSingle()*360, 
			rand.NextSingle()*360,
			rand.NextSingle()*360);
	}
	current = Quat.Slerp(current, destination, 16 * Time.Stepf);
	Mesh.Cube.Draw(Material.Default, Matrix.TRS(new Vec3(0, 0, -0.5f), current, 0.1f));
});
```

And here's what it looks like in action:
![SKSlerp](https://user-images.githubusercontent.com/8484884/234667744-168c7a99-fc26-46ef-9d53-2eb7b4c5db99.gif)

**Comment by laultman** [comment]
Thanks! Now it makes sense.
